### ISSUE TITLE
Update feature/gdas-validation to be consistent with g-w and GDASApp development

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/JEDI-T2O#107]
JEDI-T2O `feature/gdas-validation` has fallen behind g-w `develop` and GDASApp `feature/gdas-validation`.  This issue is opened to bring `feature/gdas-validation` up to date with the current  (as of 1/18/2024) state of g-w and GDASApp.

**Comment by RussTreadon-NOAA** [comment]
Update `setup_workspace.sh` as follows:
- no longer checkout GDASApp `feature/gdas-validation`.  Instead use the GDASApp hash cloned by global-workflow.  This hash is a snapshot of GDASApp `develop`.   GSI `feature/gdas-validation` is still checked out as part of `setup_workspace.sh -c`.
- updates required by changes to global-workflow `develop`.   These changes are made under the `-s` option of `setup_workspace.sh`
   - source `module_gwsetup` prior to executing workflow setup scripts 
   - replace `RESDET` with `RESDETATMOS`
   - replace `COMROT` with `COMROOT`
 - replace `3dvar_dripcg.yaml` with `3dvar_drpcg.yaml`

An additional change was made to `config.resources`.   The global-workflow `config.resources` does not include the option to run the `anal` job at C768.  GDAS-validation runs `gsi.x` at C768.   Extra logic is added to the `anal` section of `config.resources` to permit a C768 analysis job.

The updated `setup_workspace.sh` was tested on Orion.  The `-c`, `-b`, and `-s` options run to completion.

The above changes were committed at a13a12f.

**Comment by RussTreadon-NOAA** [comment]
**Orion test**

Both GSI and JEDI g-w jobs for GDAS validation were executed for 2021080100.   The observation processing, analysis, and post analysis processing jobs successfully ran to completion.  A few hiccups, noted below, had to be worked through to achieve this result.  

_GSI_
While running GSI jobs, I was reminded that the `-b` option of `setup_workspace.sh` only builds GSI and GDASApp.  Other g-w components are not built.  I manually booted `gdassfcanl` after `gdasanal` completed.  Job `gdassfcanl` failed because executable `global_cycle` did not exist.  If a developer wants to run non-GSI jobs (e.g., `gdassfcanl`) as part of GDAS validation, they should execute `sorc/build_all.sh` to build and install the full suite of g-w apps.

_GDASApp_
While running GDASApp jobs, I was reminded of three items:
1. The first submission of `gdasprepatmiodaobs` failed because `bufr2ioda_subpfl_argo_profiles.py` could not find input dump files.  We do not maintain subpfl_argo_profiles in the GDA.  Logic in GDASApp `ush/ioda/bufr2ioda/run_bufr2ioda.py` executes all converters named `bufr2ioda_*.py`.  Hence, execution of `bufr2ioda_subpfl_argo_profiles.py`.   I got around this in my test by renaming this script as `skip_bufr2ioda_subpfl_argo_profiles.py`.  This, obviously, is not the correct solution.  A better solution would be for `run_bufr2ioda.py` to read a yaml file listing the observation types to process or the converters to run.  This can be explored in a future GDASApp issue and PR.

2. Developers may need to prime the system depending on the observation type being tested.  For example, my test processed `atms_n20`.  The first run of `gdasatmanlinit` failed because bias correction files for `atms_n20` were not found in the previous cycle `analysis/atmos` directory.  I manually created this directory and copied `atms_n20` bias correction files to the appropriate location.

3. `gdasatmanlfinal` failed to include all diagnostic files in `gdas.t00z.atmstat`.  Logic in g-w `ush/python/pygfs/task/atm_analysis.py` only processes observation specific diagnostic files satisfying the pattern `diag*nc4`.  Some observation yamls use suffix `.nc` for diagnostic files.  `atm_analysis.py` fails to include `*.nc` diagnostic files in the `atmstat` tarball.  We need to standardize the netcdf suffix used for diagnostic files across all observation types.
